compiler/rustc_mir/src/transform/deduplicate_blocks.rs

```diff
@@ -0,0 +1,352 @@
+//! This pass finds basic blocks that are completely equal,
+//! and replaces all uses with just one of them.
+
+use std::{borrow::Cow, fmt::Debug, hash::Hash, hash::Hasher, iter::from_fn, iter::Peekable};
+
+use crate::transform::MirPass;
+
+use itertools::Itertools;
+use rustc_data_structures::{fx::FxHashMap, fx::FxHasher};
+use rustc_middle::mir::visit::MutVisitor;
+use rustc_middle::mir::*;
+use rustc_middle::ty::TyCtxt;
+
+use super::simplify::simplify_cfg;
+
+pub struct DeduplicateBlocks {
+    label: String,
+}
+
+impl DeduplicateBlocks {
+    pub fn new(label: &str) -> DeduplicateBlocks {
+        DeduplicateBlocks { label: format!("DeduplicateBlocks-{}", label) }
+    }
+}
+
+impl<'tcx> MirPass<'tcx> for DeduplicateBlocks {
+    fn name(&self) -> Cow<'_, str> {
+        Cow::Borrowed(&self.label)
+    }
+
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        debug!("Running DeduplicateBlocks on `{:?}`", body.source);
+        let duplicates = find_duplicates(body);
+        let should_cleanup = !duplicates.is_empty();
+        let mut opt_applier = OptApplier { tcx, duplicates };
+        opt_applier.visit_body(body);
+
+        if should_cleanup {
+            simplify_cfg(body);
+        }
+    }
+}
+
+/// Basic blocks are equal iff
```

Comment:
```suggestion
/// Basic blocks are equal if
```

Reply:
In case you didn't know, "iff" is short for [if and only if](https://en.wikipedia.org/wiki/If_and_only_if). In that light, I think the below statement is correct, is it not?